bezier-rs/lib/src/lib.rs

```diff
@@ -536,6 +536,40 @@ impl Bezier {
 			.filter(|&point| utils::dvec2_approximately_in_range(point, min, max, max_abs_diff).all())
 			.collect::<Vec<DVec2>>()
 	}
+
+	/// Returns a list of lists of points representing the hull points for all iterations at the point corresponding to `t` using de Casteljau's algorithm
+	/// the ith element of the list represents the points of the ith iteration
+	/// More information on the algorithm can be found in the [de Casteljau section in Pomax's Bezier Primer](https://pomax.github.io/bezierinfo/#decasteljau)
+	pub fn hull(&self, t: f64) -> Vec<Vec<DVec2>> {
+		let mut current_points = Vec::new();
+		let mut hull_points = Vec::new();
+		current_points.push(self.start);
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				current_points.push(handle);
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				current_points.push(handle_start);
+				current_points.push(handle_end);
+			}
+		}
+		current_points.push(self.end);
+		hull_points.push(current_points.clone());
+
+		while current_points.len() > 1 {
+			let mut next_points = Vec::new();
+			// iterate until one point is left, that point will be equal to compute(t)
```

Comment:
Is it convention that the last point is not considered part of the hull?

Reply:
When it gets down to one point, that point is already added to the hull at the end of the previous iteration.